web/app.py

```diff
@@ -96,8 +96,14 @@ def crawl():
             message = f"backing off for {domain}"
             return {"error": {"message": message}}, 429
 
+    retry_duration = 0
     try:
         response = domain_http_client.get(url, headers=headers, timeout=5)
+        if not response.ok and "Retry-After" in response.headers:
```

Comment:
This logic -- or equivalent -- should also be added to the `resolve` method.

In addition: as noted in the linked issue description, it is valid for redirects (HTTP 3xx) to specify a `Retry-After` header in the response.  Those responses are probably considered successful (`response.ok` is true), so we may need to handle those cases too (in both methods).